### ISSUE TITLE
disk space on sdr-api-prod occasionally spikes and triggers nagios alerts

### ISSUE THREAD

**Issue by jmartin-sul** [sul-dlss/sdr-api#395]
from a slack thread:

@tallenaz
> I'm seeing disk space on sdr-api-prod occasionally spike and then go back down after several minutes. It's spiking just enough to trigger a nagios alert which then clears once the spike goes down. This is happening on /sdr-storage -- can the directory be pruned at all, or should I expand the volume enough to ensure nagios quiets down?

@jcoyne
> Expand it.  That happens because files are uploaded there and then transferred on.
>
> Yeah, there’s definitely a bunch of old files in there.  We should schedule some time to take a look at how we can clean them up.  We probably shouldn’t just find old files and delete them, because they probably have database records that should be cleared at the same time.
>
> Example of an old file: `ls -lh /sdr-storage/00/9x`
> ```
> total 128K
> -rw-rw-r-- 1 sdr sdr 121K Oct 10 05:32 009xes97an9s1c64x0f8m6lg00eu
> ```

**Comment by tallenaz** [comment]
Request RITM00270734 is in to UIT to add 500G of space to the volume

**Comment by jmartin-sul** [comment]
@tallenaz adds:
> It's not a big deal on Ops' end -- UIT is expanding the volume which will quiet nagios down, and in the mean time, the spikes are well within the current capacity of the volume -- It typically spikes from 79% to 81%

**Comment by andrewjbtw** [comment]
I wonder if Google deposits that have failed have left behind files, even if they succeed later on retry.

**Comment by justinlittman** [comment]
I thought we cleaned these up: https://github.com/sul-dlss/sdr-api/blob/main/config/schedule.rb#L16

**Comment by jcoyne** [comment]
@justinlittman yes, that was my thought to.  It doesn't look like it works as well as we thought.

**Comment by tallenaz** [comment]
nagios started alerting again on this yesterday morning -- it's currently using 83% of the space (2.9 of 3.5TB), up 3% from when the alert started.

**Comment by justinlittman** [comment]
Occasional spiking is to be expected.

The storage is currently at 2% usage, so our cleanup strategy seems to be working:
```
$ df -h storage
Filesystem                   Size  Used Avail Use% Mounted on
sf5-webapp:/sdr_api_storage  5.0T   59G  5.0T   2% /opt/app/sdr/sdr-api/shared/storage
```

I've also reviewed the logs from the cleanup cron and the related code.

Closing for now; can be re-opened if continues to be problematic.
